2-repositories/iglu-server/src/main/scala/com/snowplowanalytics/iglu/server/service/SchemaService.scala

```diff
@@ -127,13 +135,31 @@ class SchemaService[F[+_]: Sync](swagger: SwaggerSyntax[F],
               existing <- db.getSchema(schema.self).map(_.isDefined)
               _        <- db.addSchema(schema.self, schema.schema, isPublic)
               payload   = IgluResponse.SchemaUploaded(existing, schema.self.schemaKey): IgluResponse
+              _        <- schemaPublishedWebhooks.map(sendSchemaPublishedWebhook(_, schema.self.schemaKey, existing))
               response <- if (existing) Ok(payload) else Created(payload)
             } yield response
           case Left(error) =>
             Conflict(IgluResponse.Message(error): IgluResponse)
         }
       } yield response
     else Forbidden(IgluResponse.Forbidden: IgluResponse)
+
+  private def sendSchemaPublishedWebhook(webhook: Config.SchemaPublishedWebhook, schema: SchemaKey, existing: Boolean) = {
+    Uri.fromString(webhook.url) match {
```

Comment:
Again, why we're checking validity of `URI` only here? Why whould anyone run an improperly configured server only to get these errors.

"Fail fast" is a very important software engineering practice: https://www.martinfowler.com/ieeeSoftware/failFast.pdf. Especially when you have a statically typed language.